### ISSUE TITLE
请问在训练的时候，指标的数值一直很高，该如何解决呢？

### ISSUE THREAD

**Issue by Yhc-777** [Shimmer93/MPCount#12]
Val criterion: 218.1800 mse: 161874.7692 best: 224.6491, time: 168.9916

你好打扰啦，请问我在训练qnrf和sta、stb的时候，print出来的指标总是很高，尝试下调整了超参数但是还是变化不大，请问我应该如何解决呢？谢谢！

**Comment by Shimmer93** [comment]
可以发一下训练的script、config file和log吗？

**Comment by Yhc-777** [comment]
[log.txt](https://github.com/user-attachments/files/16622956/log.txt)


script：python main.py --task train --config configs/sta_train.yml

config：

seed: 2023
version: sta
device: 'cuda:0'
log_para: 1000
patch_size: 10000
mode: 'final'
num_epochs: 180
# checkpoint: null
checkpoint: /mnt/2/hc/MPCount/pretrained_models/sta.pth
model:
  name: 'final'
  params:
    pretrained: True
    mem_size: 1024
    mem_dim: 256
    cls_thrs: 0.5
    err_thrs: 0.5
    den_dropout: 0.5
    cls_dropout: 0.3
    has_err_loss: False
train_dataset: &train_dataset_params
  name: 'den_cls'
  params: 
    root: 'data/sta'
    crop_size: 320
    downsample: 1
    is_grey: False
    unit_size: 16
    pre_resize: 1
val_dataset: *train_dataset_params
test_dataset:
  name: 'den_cls'
  params: 
    root: '/mnt/2/hc/MPCount/data/ShanghaiTech/part_B/test_data/images'
    crop_size: 320
    downsample: 1
    is_grey: False
    unit_size: 16
    pre_resize: 1
train_loader:
  batch_size: 8
  num_workers: 8
  shuffle: True
  pin_memory: True
val_loader: &val_loader_params
  batch_size: 1
  num_workers: 4
  shuffle: False
  pin_memory: False
test_loader: *val_loader_params
optimizer:
  name: 'adamw'
  params:
    lr: &lr 0.001
    weight_decay: 0.0001
scheduler:
  name: 'onecycle'
  params:
    max_lr: *lr
    epochs: 300
    steps_per_epoch: 15
    final_div_factor: 1000
    
    
    辛苦帮忙查看下！谢谢！！！

**Comment by Shimmer93** [comment]
看起来没有问题，可以检查下testing的script和config file

**Comment by Yhc-777** [comment]
testing的script:
python main.py --task test --config configs/sta_test_stb.yml

config file:保持和repo的一致没有改动。

刚刚又测试了下，效果是：

(mpcount) u@u:/mnt/2/hc/MPCount$ python main.py --task test --config configs/sta_test_stb.yml
Start testing at 2024-08-15 16:24:27
Loading checkpoint from logs/sta/best.pth
Testing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:37
Testing results: mae: 13.0754 mse: 574.5333 
Testing results saved to logs/sta
End testing at 2024-08-15 16:25:05


这个的效果是属于较好的吗？麻烦你啦实在是！

**Comment by Shimmer93** [comment]
这个效果没问题

**Comment by Yhc-777** [comment]
明白，辛苦啦！谢谢！

**Comment by kexiaolong0121** [comment]
你好，可以分享一下你STB训练的脚本参数吗，我的结果始终差的远一些
